src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
```

Comment:
PullProofAI

Consider adding validation for the presence of `x-feedback-secret` header before comparing it with the secret.
```typescript
if (!requestSecret) {
    throw new UnauthorizedException('Feedback secret missing');
}

if (requestSecret === secret) {
    return true;
} else {
    throw new UnauthorizedException('Invalid feedback secret');
}
```
By explicitly checking for the existence of the `x-feedback-secret` header, you can provide a more specific error message when the header is missing.


<!-- This is an auto-generated comment by PullProofAI -->